apps/transport/lib/transport/data_checker.ex

```diff
@@ -21,45 +21,69 @@ defmodule Transport.DataChecker do
   def inactive_data do
     # Some datasets marked as inactive in our database may have reappeared
     # on the data gouv side, we'll mark them back as active.
-    to_reactivate_datasets = get_to_reactivate_datasets()
-    reactivated_ids = Enum.map(to_reactivate_datasets, & &1.datagouv_id)
+    datasets_statuses = datasets_datagouv_statuses()
+
+    to_reactivate_datasets =
+      datasets_statuses
+      |> Enum.filter(fn {%Dataset{is_active: is_active}, status} -> not is_active and status == :active end)
+      |> Enum.map(fn {%Dataset{} = dataset, _status} -> dataset end)
```

Comment:
```suggestion
    for {%Dataset{is_active: false} = dataset, :active} <- datasets_statuses, do: dataset
```

Reply:
une boucle `for` est élégante dans ce cas je trouve, ça permet de filtrer en pattern matchant